src/deepsparse/benchmark_model/benchmark_model.py

```diff
@@ -0,0 +1,255 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import os
+
+from deepsparse import Scheduler, compile_model
+from deepsparse.benchmark_model.benchmark import model_stream_benchmark
+from deepsparse.utils import (
+    generate_random_inputs,
+    model_to_path,
+    override_onnx_input_shapes,
+    parse_input_shapes,
+)
+from deepsparse.utils.log import log_init
+
+
+log = log_init(os.path.basename(__file__))
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Benchmark ONNX models in the DeepSparse Engine"
+    )
+
+    parser.add_argument(
+        "model_path",
+        type=str,
+        help="Path to an ONNX model file or SparseZoo model stub",
+    )
+
+    parser.add_argument(
+        "-b",
+        "--batch_size",
+        type=int,
+        default=1,
+        help="The batch size to run the analysis for. Must be greater than 0",
+    )
+    parser.add_argument(
+        "-shapes",
+        "--input_shapes",
+        type=str,
+        default="",
+        help="Override the shapes of the inputs, "
+        'i.e. -shapes "[1,2,3],[4,5,6],[7,8,9]" results in '
+        "input0=[1,2,3] input1=[4,5,6] input2=[7,8,9]",
+    )
+    parser.add_argument(
+        "-ncores",
+        "--num_cores",
+        type=int,
+        default=None,
+        help=(
+            "The number of physical cores to run the analysis on, "
+            "defaults to all physical cores available on the system"
+        ),
+    )
+    parser.add_argument(
+        "-s",
+        "--scenario",
+        type=str,
+        default="async",
+        choices=["async", "sync"],
+        help=(
+            "Choose between using a sync/async scenarios. This is similar to the "
+            "single-stream/multi-stream scenarios. Default value is async."
+        ),
+    )
+    parser.add_argument(
+        "-t",
+        "--time",
+        type=int,
+        default=10,
+        help="The number of seconds the benchmark will run. Default is 10 seconds.",
+    )
+    parser.add_argument(
+        "-nstreams",
+        "--num_streams",
+        type=int,
+        default=None,
+        help=(
+            "The number of streams that will submit inferences in parallel using "
+            "async scenario. Default is automatically determined for given hardware "
+            "and may be sub-optimal."
+        ),
+    )
+    parser.add_argument(
+        "-pin",
+        "--thread_pinning",
+        type=str,
+        default="true",
+        choices=["true", "false", "numa"],
```

Comment:
would "none", "cores", "numa" be more descriptive names?

Reply:
i like that, will change